controllers/operatorconfigmap_controller.go

```diff
@@ -170,209 +172,249 @@ func (c *OperatorConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return reconcile.Result{}, err
 	}
 
-	if err := c.reconcileSubscriptionValidatingWebhook(); err != nil {
-		c.log.Error(err, "unable to register subscription validating webhook")
-		return ctrl.Result{}, err
-	}
-
-	if err := labelClientOperatorSubscription(c); err != nil {
-		c.log.Error(err, "unable to label ocs client operator subscription")
-		return ctrl.Result{}, err
-	}
-
-	if err := c.reconcileSubscription(); err != nil {
-		c.log.Error(err, "unable to reconcile subscription")
-		return ctrl.Result{}, err
-	}
+	if c.operatorConfigMap.GetDeletionTimestamp().IsZero() {
 
-	if err := c.ensureConsolePlugin(); err != nil {
-		c.log.Error(err, "unable to deploy client console")
-		return ctrl.Result{}, err
-	}
-
-	if deployCSI, err := c.getDeployCSIConfig(); err != nil {
-		c.log.Error(err, "failed to perform precheck for deploying CSI")
-		return ctrl.Result{}, err
-	} else if deployCSI {
-		clusterVersion := &configv1.ClusterVersion{}
-		clusterVersion.Name = clusterVersionName
-		if err := c.get(clusterVersion); err != nil {
-			c.log.Error(err, "failed to get the clusterVersion version of the OCP cluster")
-			return reconcile.Result{}, err
+		//ensure finalizer
+		if !controllerutil.ContainsFinalizer(c.operatorConfigMap, OperatorConfigMapFinalizer) {
+			c.log.Info("finalizer missing on the operatorConfigMap resource, adding...")
+			controllerutil.AddFinalizer(c.operatorConfigMap, OperatorConfigMapFinalizer)
```

Comment:
if your intention is to just add finalizer why can't the response (bool) of AddFinalizer be used directly?

Reply:
There is no real reason other than readability of the code. When i was writing the commit i was drawing inspiration from both the code base already existing here and the osd-deployer (there it's not exactly the same though).
This can be changed.